hazelcast-jet-s3/src/main/java/com/hazelcast/jet/s3/S3Sinks.java

```diff
@@ -0,0 +1,167 @@
+     *                     .withRegion(Regions.US_EAST_1)
+     *                     .build();
+     *         },
+     *         Object::toString
+     * );
+     * Pipeline p = Pipeline.create();
+     * p.drawFrom(Sources.map("map", alwaysTrue(), Map.Entry::getValue))
+     *  .drainTo(sink);
+     * }</pre>
+     *
+     * @param bucketName     the name of the bucket
+     * @param linesPerFile   the number of lines per file
+     * @param clientSupplier S3 client supplier
+     * @param toStringFn     the function which converts each item to its
+     *                       string representation
+     * @param <T>            type of the items the sink accepts
+     */
+    @Nonnull
+    public static <T> Sink<? super T> s3(
+            @Nonnull String bucketName,
+            int linesPerFile,
+            @Nonnull SupplierEx<? extends AmazonS3> clientSupplier,
+            @Nonnull FunctionEx<? super T, String> toStringFn
+    ) {
+        return SinkBuilder
+                .sinkBuilder("s3-sink", context ->
+                        new S3Context<>(bucketName, context.globalProcessorIndex(),
+                                linesPerFile, clientSupplier, toStringFn))
+                .<T>receiveFn(S3Context::receive)
+                .destroyFn(S3Context::close)
+                .build();
+    }
+
+    private static final class S3Context<T> {
+
+        private final String bucketName;
+        private final int processorIndex;
+        private final int linesPerFile;
+        private final AmazonS3 amazonS3;
+        private final FunctionEx<? super T, String> toStringFn;
+
+        private int itemCounter;
+        private long objectCounter;
+        private StringBuilder buffer = new StringBuilder();
+
+        private S3Context(
+                String bucketName,
+                int processorIndex,
+                int linesPerFile,
+                SupplierEx<? extends AmazonS3> clientSupplier,
+                FunctionEx<? super T, String> toStringFn
+        ) {
+            this.bucketName = bucketName;
+            this.processorIndex = processorIndex;
+            this.linesPerFile = linesPerFile;
+            this.amazonS3 = clientSupplier.get();
+            this.toStringFn = toStringFn;
+
+            checkIfBucketExists();
+        }
+
+        private void checkIfBucketExists() {
+            if (!amazonS3.doesBucketExistV2(bucketName)) {
+                throw new IllegalArgumentException("Bucket [" + bucketName + "] does not exist");
+            }
+        }
+
+        private void receive(T item) {
+            buffer.append(toStringFn.apply(item))
+                  .append(System.lineSeparator());
+            if (++itemCounter == linesPerFile) {
+                amazonS3.putObject(bucketName, nextKey(), buffer.toString());
+                buffer.setLength(0);
+                itemCounter = 0;
+            }
+        }
+
+        private void close() {
```

Comment:
`close()` is called after job is completed, but here we are writing. I'm not sure if this is a good idea.